src/test/java/com/epam/jamp2/service/impl/CalculationCommandExecutionServiceImplTest.java

```diff
@@ -0,0 +1,127 @@
+		USD2CNYRate.put("CNY", new BigDecimal("7.00"));
+		response.setRates(USD2CNYRate);
+		@SuppressWarnings("unchecked")
+		Call<FixerioResponse> call = mock(Call.class);
+	    given(call.execute()).willReturn(Response.success(response));
+		given(fixerioServiceProxy.getRates("USD")).willReturn(call);
+	}
+
+	@Test
+	public void testShouldNotCallForRateWhenInputCcysAreSameAsTargetCcy() throws CommandFormatException, IOException {
+
+		 CalculationCommand calculationCommand = CalculationCommand.parseFromString("CNY7.00+CNY1.00=CNY");
+		 calculationCommandExecutionService.calculate(calculationCommand);
+		 verify(fixerioServiceProxy, times(0)).getRates(anyObject());
+
+	}
+	
+	@Test
+	public void testShouldCallForRateForOneTimeWhenInputCcysHasOneSameAsTargetCcy() throws CommandFormatException, IOException {
+
+		 CalculationCommand calculationCommand = CalculationCommand.parseFromString("CNY7.00+USD1.00=CNY");
+		 calculationCommandExecutionService.calculate(calculationCommand);
+		 verify(fixerioServiceProxy, times(1)).getRates(anyObject());
+
+	}
+	
+	@Test
+	public void testShouldCallForRateForTwoTimesWhenInputCcysAreNotSameAsTargetCcy() throws CommandFormatException, IOException {
+
+		 CalculationCommand calculationCommand = CalculationCommand.parseFromString("USD1.00+USD2.00=CNY");
+		 calculationCommandExecutionService.calculate(calculationCommand);
+		 verify(fixerioServiceProxy, times(2)).getRates(anyObject());
+
+	}
+	
+	@Test
+	public void testShouldReturnCorrectResultWhenDoAdd() throws CommandFormatException, IOException {
+
+		 CalculationCommand calculationCommand = CalculationCommand.parseFromString("CNY7.00+USD1.00=CNY");
+		 Value value = calculationCommandExecutionService.calculate(calculationCommand);
+		 assertThat(value.getValue(), is(equalTo(new BigDecimal("14.0000"))));
+		 assertEquals(Optional.of("CNY"), value.getCurrencyCode());
+
+	}
+	
+	@Test
+	public void testShouldReturnCorrectResultWhenDoMultiply() throws CommandFormatException, IOException {
+
+		 CalculationCommand calculationCommand = CalculationCommand.parseFromString("CNY7.00*USD1.00=CNY");
+		 Value value = calculationCommandExecutionService.calculate(calculationCommand);
+		 assertThat(value.getValue(), is(equalTo(new BigDecimal("49.000000"))));
+		 assertEquals(Optional.of("CNY"), value.getCurrencyCode());
+
+	}
+	
+	@Test
+	public void testShouldReturnCorrectResultWhenDoDivide() throws CommandFormatException, IOException {
+
+		 CalculationCommand calculationCommand = CalculationCommand.parseFromString("CNY7.00/USD1.00=CNY");
+		 Value value = calculationCommandExecutionService.calculate(calculationCommand);
+		 assertThat(value.getValue(), is(equalTo(new BigDecimal("1"))));
+		 assertEquals(Optional.of("CNY"), value.getCurrencyCode());
+
+	}
+	
+	@Test(expected = CommandFormatException.class)
+	public void testShouldThrowCommandFormatExceptionWhenFormatInvalid() throws CommandFormatException, IOException {
```

Comment:
I would also recommend to have separate test for positive parsing scenario even though you implicitly check it in other tests of this class